packages/opencensus-core/src/stats/model/types.ts

```diff
@@ -0,0 +1,148 @@
+   */
+  getRegistredViews(): RegistredViews;
+  /**
+   * Creates a new aggregation instance
+   * @param bucketBoundaries
+   */
+  createAggregation(bucketBoundaries: BucketBoundaries): Aggregation;
+  /**
+   * Records values to measurements
+   * @param measurements
+   */
+  record(...measurements: Measurement[]): void;
+}
+
+export interface Tags { [key: string]: string; }
+
+export enum MeasureType {
+  /** Unknown type. */
+  TYPE_UNSPECIFIED = 'TYPE_UNSPECIFIED',
+  /** Indicates an int64 Measure. */
+  INT64 = 'INT64',
+  /** Indicates a double Measure. */
+  DOUBLE = 'DOUBLE'
+}
+
+export interface Measure {
+  /**
+   * A string by which the measure will be referred to, e.g.
+   * "rpc_server_latency". Names MUST be unique within the library.
+   */
+  name: string;
+  /** Describes the measure, e.g. "RPC latency in seconds". */
+  description: string;
+  /**
+   * Describes the unit used for the Measure. Follows the format described by
+   * http://unitsofmeasure.org/ucum.html.
+   */
+  unit: string;
+  /** The type used for this Measure. */
+  type: MeasureType;
+}
+
+/** View path from view name */
+export interface ViewPath { [path: string]: ViewPath; }
+
+export interface View {
+  /**
+   * A string by which the View will be referred to, e.g. "rpc_latency". Names
+   * MUST be unique within the library.
+   */
+  name: string;
+  /** Describes the view, e.g. "RPC latency distribution" */
+  description: string;
+  /** The Measure to which this view is applied. */
+  measure: Measure;
+  /**
+   * An array of tags. These values associated with tags of this name form the
+   * basis by which individual stats will be aggregated (one aggregation per
+   * unique tag value). If none are provided, then all data is recorded in a
+   * single aggregation.
+   */
+  columns: Tags;
+  /** An Aggregation describes how data collected is aggregated */
+  aggregation: Aggregation;
+  /**
+   * Gets view path from view name. Each folder is separated by '/' in the view
+   * name
+   */
+  getPath(): ViewPath;
+}
+
+export interface Aggregation {
+  /**
+   * Counts the number of measurements recorded.
+   */
+  count(): number;
+  /**
+   * Indicates that data collected and aggregated with this Aggregation will be
+   * summed up.
+   */
+  sum(): number;
+  /**
+   * Indicates that data collected and aggregated with this Aggregation will
+   * represent the last recorded value.
+   */
+  lastValue(): number;
+  /**
+   * Indicates that the desired Aggregation is a histogram distribution. A
+   * distribution Aggregation may contain a histogram of the values in the
+   * population.
+   */
+  distribution(): Histogram[];
+}
+
+/** Describes a data point to be collected for a Measure. */
+export interface Measurement {
+  tags: Tags;
+  /** The measure to which the value is applied. */
+  measure: Measure;
+  /** The recorded value */
+  value: number;
+}
+
+/** Sorted number array */
+export interface BucketBoundaries {
```

Comment:
Why does this need to be a type + implementation at all? Can't we just use a list of numbers?

Reply:
Interface removed, now using just number[]